File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int k1 = prices[0];
        int k2 = 0;

        for(int i = 1;i < prices.length;i++)
        {
            int curr = prices[i] - k1;

            if(curr > k2)
            {
                k2 = curr;
            }

            if(k1 > prices[i])
            {
                k1 = prices[i];
            }
        }
        return k2; 
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        p = 0
        c = prices[0]

        for i in range(1,len(prices)):
            if p < prices[i] - c:
                p = prices[i] - c

            if c > prices[i]:
                c = prices[i]

        return p